### PR TITLE
tidb-functions: update tidb-functions.md doc

### PR DESCRIPTION
**Pull request by Oreoxmt**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v6.1 (TiDB 6.1 versions)
- [x] v6.0 (TiDB 6.0 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/8775 https://github.com/pingcap/docs-cn/pull/9632
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: functions-and-operators/tidb-functions.md [hunk 1]
```diff
@@ -230,3 +231,69 @@ See also:
...
+    When there is a hotspot caused by monotonically increasing or decreasing data on the index field `a` in the unique secondary index `uk((tidb_shard(a)), a))`, the index's prefix `tidb_shard(a)` can scatter the hotspot to improve the cluster's scalability.
+
+- Scenario:
+
+    - When there is a write hotspot caused by increasing or decreasing keys on the unique secondary index, and the index contains integer type fields.
+    - The SQL statement executes an equality query based on all fields of the secondary index, either as a separate `SELECT` or as an internal query generated by `UPDATE`, `DELETE`, etc. The equality query includes `a = 1` or `a IN (1, 2, ......)` these two ways.
+
+- Limitations:
+
+    - SHARD INDEX cannot be used with inequality or `IN` queries.
```
Comment (Oreoxmt) [inline functions-and-operators/tidb-functions.md:252]:
```suggestion
    - SHARD INDEX cannot be used with inequality queries.
```

File: functions-and-operators/tidb-functions.md [hunk 2]
```diff
@@ -230,3 +231,69 @@ See also:
...
+- Scenario:
+
+    - When there is a write hotspot caused by increasing or decreasing keys on the unique secondary index, and the index contains integer type fields.
+    - The SQL statement executes an equality query based on all fields of the secondary index, either as a separate `SELECT` or as an internal query generated by `UPDATE`, `DELETE`, etc. The equality query includes `a = 1` or `a IN (1, 2, ......)` these two ways.
+
+- Limitations:
+
+    - SHARD INDEX cannot be used with inequality queries.
+    - SHARD INDEX cannot be used with an outmost `AND` operator mixed with `OR`.
+    - SHARD INDEX cannot be used with `GROUP BY` clause.
```
Comment (TonsnakeLin) [inline functions-and-operators/tidb-functions.md:254]:
```suggestion
    - SHARD INDEX cannot be used in`GROUP BY` clause.
```

File: functions-and-operators/tidb-functions.md [hunk 3]
```diff
@@ -230,3 +231,69 @@ See also:
...
+    - SHARD INDEX cannot be used with `ORDER BY` clause.
+    - SHARD INDEX cannot be used with the `ON` clause.
```
Comment (TonsnakeLin) [inline functions-and-operators/tidb-functions.md:256]: change 'with' to 'in'

File: functions-and-operators/tidb-functions.md [hunk 4]
```diff
@@ -17,6 +17,7 @@ The following functions are TiDB extensions, and are not present in MySQL:
 | [`TIDB_VERSION()`](#tidb_version) | The `TIDB_VERSION` function returns the TiDB version with additional build information. |
 | [`TIDB_DECODE_SQL_DIGESTS(digests, stmtTruncateLength)`](#tidb_decode_sql_digests) | The `TIDB_DECODE_SQL_DIGESTS()` function is used to query the normalized SQL statements (a form without formats and arguments) corresponding to the set of SQL digests in the cluster. |
 | `VITESS_HASH(str)` | The `VITESS_HASH` function returns the hash of a string that is compatible with Vitess' `HASH` function. This is intended to help the data migration from Vitess. |
+| `TIDB_SHARD()` | The `TIDB_SHARD` function can be used to create a SHARD INDEX to scatter the index hotspot. A SHARD INDEX is an expression index with a `TIDB_SHARD`function as the prefix.|
```
Comment (TomShawn) [inline functions-and-operators/tidb-functions.md:20]: If SHARD INDEX is natural language, shall we use the lower case "shard index"? Or wrap it with backticks.
```suggestion
| `TIDB_SHARD()` | The `TIDB_SHARD` function can be used to create a SHARD INDEX to scatter the index hotspot. A SHARD INDEX is an expression index with a `TIDB_SHARD` function as the prefix.|
```

File: functions-and-operators/tidb-functions.md [hunk 5]
```diff
@@ -230,3 +231,69 @@ See also:
...
+
+### TIDB_SHARD
+
+The `TIDB_SHARD` function can be used to create a SHARD INDEX to scatter the index hotspot. A SHARD INDEX is an expression index with a `TIDB_SHARD`function as the prefix.
+
+#### SHARD INDEX
+
+- Creation:
+
+    When there is a hotspot caused by monotonically increasing or decreasing data on the index field `a` in the unique secondary index `uk((tidb_shard(a)), a))`, the index's prefix `tidb_shard(a)` can scatter the hotspot to improve the cluster's scalability.
```
Comment (TomShawn) [inline functions-and-operators/tidb-functions.md:243]: See https://github.com/pingcap/docs-cn/pull/8775/files#r890124259

File: functions-and-operators/tidb-functions.md [hunk 6]
```diff
@@ -230,3 +231,69 @@ See also:
...
+
+- Scenario:
```
Comment (TomShawn) [inline functions-and-operators/tidb-functions.md:245]:
```suggestion
- Scenarios:
```

File: functions-and-operators/tidb-functions.md [hunk 7]
```diff
@@ -230,3 +231,69 @@ See also:
...
+
+    - When there is a write hotspot caused by monotonically increasing or decreasing keys on the unique secondary index, and the index contains integer type fields.
+    - The SQL statement executes an equality query based on all fields of the secondary index, either as a separate `SELECT` or as an internal query generated by `UPDATE`, `DELETE` and so on. The equality query includes `a = 1` or `a IN (1, 2, ......)` these two ways.
+
+- Limitations:
+
+    - SHARD INDEX cannot be used in inequality queries.
```
Comment (TomShawn) [inline functions-and-operators/tidb-functions.md:252]:
```suggestion
    - SHARD INDEX cannot be used in inequality or `IN` queries.
```

File: functions-and-operators/tidb-functions.md [hunk 8]
```diff
@@ -230,3 +231,69 @@ See also:
...
+    - SHARD INDEX cannot be used in an outmost `AND` operator mixed with `OR`.
+    - SHARD INDEX cannot be used in `GROUP BY` clause.
+    - SHARD INDEX cannot be used in `ORDER BY` clause.
+    - SHARD INDEX cannot be used in the `ON` clause.
```
Comment (TomShawn) [inline functions-and-operators/tidb-functions.md:256]:
```suggestion
    - SHARD INDEX cannot be used in the `GROUP BY` clause.
    - SHARD INDEX cannot be used in the `ORDER BY` clause.
    - SHARD INDEX cannot be used in the `ON` clause.
```

File: functions-and-operators/tidb-functions.md [hunk 9]
```diff
@@ -230,3 +231,69 @@ See also:
...
+    - SHARD INDEX cannot be used in the `WHERE` subquery.
+    - SHARD INDEX can only scatter unique indexes of integer fields.
+    - SHARD INDEX might be ineffective in composite indexes.
```
Comment (TomShawn) [inline functions-and-operators/tidb-functions.md:259]:
```suggestion
    - SHARD INDEX might take effect in composite indexes.
```

File: functions-and-operators/tidb-functions.md [hunk 10]
```diff
@@ -230,3 +231,69 @@ See also:
...
+    ```sql
+    +-------------------------+
+    | TIDB_SHARD(12373743746) |
+    +-------------------------+
+    |                     184 |
+    +-------------------------+
+    1 row in set (0.00 sec)
+    ```
+
+- Create a SHARD INDEX with the `TIDB_SHARD` function:
```
Comment (TomShawn) [inline functions-and-operators/tidb-functions.md:293]:
```suggestion
- Create a SHARD INDEX using the `TIDB_SHARD` function:
```

**Review by TomShawn — COMMENTED** [review]
mostly minor comments

File: functions-and-operators/tidb-functions.md [hunk 11]
```diff
@@ -230,3 +231,69 @@ See also:
...
+
+### TIDB_SHARD
+
+The `TIDB_SHARD` function can be used to create a shard index to scatter the index hotspot. A shard index is an expression index prefixed with a `TIDB_SHARD` function.
+
+#### Shard index
+
+- Creation:
+
+    When there is a hotspot caused by monotonically increasing or decreasing data on the index field `a` in the unique secondary index `uk((tidb_shard(a)), a))`, the index's prefix `tidb_shard(a)` can scatter the hotspot to improve the cluster's scalability.
```
Comment (Oreoxmt) [inline functions-and-operators/tidb-functions.md:243]:
```suggestion
    You can use `uk((tidb_shard(a)), a))` to create a shard index for the index `a`. When there is a hotspot caused by monotonically increasing or decreasing data on the index field `a` in the unique secondary index `uk((tidb_shard(a)), a))`, the index's prefix `tidb_shard(a)` can scatter the hotspot to improve the cluster's scalability.
```

File: functions-and-operators/tidb-functions.md (hunk 5 above)
Comment (Oreoxmt) [reply functions-and-operators/tidb-functions.md:243]: `uk((tidb_shard(a)), a))` 就是创建的方式，我加了一句 You can use `uk((tidb_shard(a)), a))` to create a shard index for the index `a`.

File: functions-and-operators/tidb-functions.md [hunk 12]
```diff
@@ -230,3 +231,69 @@ See also:
...
+
+### TIDB_SHARD
+
+The `TIDB_SHARD` function can be used to create a shard index to scatter the index hotspot. A shard index is an expression index prefixed with a `TIDB_SHARD` function.
+
+#### Shard index
+
+- Creation:
+
+    You can use `uk((tidb_shard(a)), a))` to create a shard index for the index `a`. When there is a hotspot caused by monotonically increasing or decreasing data on the index field `a` in the unique secondary index `uk((tidb_shard(a)), a))`, the index's prefix `tidb_shard(a)` can scatter the hotspot to improve the cluster's scalability.
```
Comment (Oreoxmt) [inline functions-and-operators/tidb-functions.md:243]:
```suggestion
    You can use `uk((tidb_shard(a)), a))` to create a shard index for the field `a`. When there is a hotspot caused by monotonically increasing or decreasing data on the index field `a` in the unique secondary index `uk((tidb_shard(a)), a))`, the index's prefix `tidb_shard(a)` can scatter the hotspot to improve the cluster's scalability.
```

File: functions-and-operators/tidb-functions.md (hunk 7 above)
Comment (Oreoxmt) [reply functions-and-operators/tidb-functions.md]: Remove `IN` query in https://github.com/pingcap/docs-cn/pull/9632

File: functions-and-operators/tidb-functions.md [hunk 13]
```diff
@@ -17,6 +17,7 @@ The following functions are TiDB extensions, and are not present in MySQL:
 | [`TIDB_VERSION()`](#tidb_version) | The `TIDB_VERSION` function returns the TiDB version with additional build information. |
 | [`TIDB_DECODE_SQL_DIGESTS(digests, stmtTruncateLength)`](#tidb_decode_sql_digests) | The `TIDB_DECODE_SQL_DIGESTS()` function is used to query the normalized SQL statements (a form without formats and arguments) corresponding to the set of SQL digests in the cluster. |
 | `VITESS_HASH(str)` | The `VITESS_HASH` function returns the hash of a string that is compatible with Vitess' `HASH` function. This is intended to help the data migration from Vitess. |
+| `TIDB_SHARD()` | The `TIDB_SHARD` function can be used to create a shard index to scatter the index hotspot. A shard index is an expression index with a `TIDB_SHARD`function as the prefix.|
```
Comment (TomShawn) [inline functions-and-operators/tidb-functions.md:20]:
```suggestion
| `TIDB_SHARD()` | The `TIDB_SHARD` function can be used to create a shard index to scatter the index hotspot. A shard index is an expression index with a `TIDB_SHARD` function as the prefix.|
```

File: functions-and-operators/tidb-functions.md [hunk 14]
```diff
@@ -230,3 +231,69 @@ See also:
...
+
+### TIDB_SHARD
+
+The `TIDB_SHARD` function can be used to create a shard index to scatter the index hotspot. A shard index is an expression index prefixed with a `TIDB_SHARD` function.
+
+#### Shard index
+
+- Creation:
+
+    You can use `uk((tidb_shard(a)), a))` to create a shard index for the field `a`. When there is a hotspot caused by monotonically increasing or decreasing data on the index field `a` in the unique secondary index `uk((tidb_shard(a)), a))`, the index's prefix `tidb_shard(a)` can scatter the hotspot to improve the cluster's scalability.
```
Comment (TomShawn) [inline functions-and-operators/tidb-functions.md:243]:
```suggestion
    To create a shard index for the index field `a`, you can use `uk((tidb_shard(a)), a))`. When there is a hotspot caused by monotonically increasing or decreasing data on the index field `a` in the unique secondary index `uk((tidb_shard(a)), a))`, the index's prefix `tidb_shard(a)` can scatter the hotspot to improve the scalability of the cluster.
```

File: functions-and-operators/tidb-functions.md [hunk 15]
```diff
@@ -230,3 +231,69 @@ See also:
...
+
+- Scenarios:
+
+    - When there is a write hotspot caused by monotonically increasing or decreasing keys on the unique secondary index, and the index contains integer type fields.
```
Comment (TomShawn) [inline functions-and-operators/tidb-functions.md:247]:
```suggestion
    - There is a write hotspot caused by monotonically increasing or decreasing keys on the unique secondary index, and the index contains integer type fields.
```

File: functions-and-operators/tidb-functions.md [hunk 16]
```diff
@@ -230,3 +231,69 @@ See also:
...
+    - Shard index cannot be used in inequality queries.
+    - Shard index cannot be used in queries that contain `OR` mixed with an outmost `AND` operator.
+    - Shard index cannot be used in the `GROUP BY` clause.
+    - Shard index cannot be used in the `ORDER BY` clause.
+    - Shard index cannot be used in the `ON` clause.
+    - Shard index cannot be used in the `WHERE` subquery.
+    - Shard index can only scatter unique indexes of integer fields.
+    - Shard index might not take effect in composite indexes.
+    - Shard index cannot go through FastPlan process, which affects optimizer performance.
+    - Shard index cannot be used in preparing the execution plan cache.
```
Comment (TomShawn) [inline functions-and-operators/tidb-functions.md:261]: Because all items start with "Shard index", it is suggested to remove the common subject.

Comment (TomShawn) [reply functions-and-operators/tidb-functions.md:261]: Cannot be used to ...

**Review by TomShawn — APPROVED** [review]
rest LGTM

File: functions-and-operators/tidb-functions.md [hunk 17]
```diff
@@ -230,3 +231,69 @@ See also:
...
+
+- Limitations:
+
+    - Cannot be used in inequality queries.
+    - Cannot be used in queries that contain `OR` mixed with an outmost `AND` operator.
+    - Cannot be used in the `GROUP BY` clause.
+    - Cannot be used in the `ORDER BY` clause.
+    - Cannot be used in the `ON` clause.
+    - Cannot be used in the `WHERE` subquery.
+    - Can only be used to scatter unique indexes of integer fields.
```
Comment (TomShawn) [inline functions-and-operators/tidb-functions.md:258]:
```suggestion
    - Can be used to scatter unique indexes of only the integer fields.
```

**Pull request merged**